src/Raven.Server/ServerWide/Commands/ClusterTransactionCommand.cs

```diff
@@ -447,7 +436,7 @@ internal static long GetPrevCount(ClusterOperationContext context, Tree commands
             using (GetPrefix(context, databaseName, out var databaseSlice))
             {
                 var prevCount = commandsCountPerDatabase.ReadInt64(databaseSlice) ?? 0;
-                using (GetPrefix(context, databaseName, out var prefixSlice, prevCount))
+                using (GetPrefix(context, databaseName, out var prefixSlice, prevCount, index))
```

Comment:
so the new key format will be something like this:
`{database}/{prev}/{index}` ?

We need to consider backward comp.
- What happened when updated node doing a Snapshot (raft log snapshot) to an old node
- What happened if we persist this the old way in the raft log, but the database didn't handled that yet and we upgrade the server

Reply:
> What happened when updated node doing a Snapshot (raft log snapshot) to an old node

This is not an issue while the cluster logs are the same.
The `ClusterStateMachine.TransactionCommands` table is used only internally

>  What happened if we persist this the old way in the raft log, but the database didn't handled that yet and we upgrade the server

Good point.
I will change the code to look for the `Separator` and read the value after that as `prevCount`.
This will work only for upgrades but will not work for downgrades.
Should we mark it somehow?